apps/nowcasting-app/pages/index.tsx

```diff
@@ -4,53 +4,121 @@ import useSWR from "swr";
 
 import Layout from "../components/layout";
 import Map from "../components/map";
+import { API_PREFIX } from "../constant";
 import ButtonGroup from "../components/button-group";
+import gspShapeData from "../data/gsp-regions.json";
+import * as d3 from "d3";
+
 
 const fetcher = (input: RequestInfo, init: RequestInit) =>
   fetch(input, init).then((res) => res.json());
 
-const API_PREFIX = "https://api-dev.nowcasting.io/v0";
+// Assuming first item in the array is the latest
+const latestForecastValue = 0;
 
 export default function Home() {
-  const { data: forecastData, error: forecastError } = useSWR(
+
+  const { data: initForecastData, error: forecastError, mutate } = useSWR(
     `${API_PREFIX}/GB/solar/gsp/forecast/all`,
-    fetcher
+    fetcher,
   );
 
   if (forecastError) {
     console.log(forecastError);
     return <div>Failed to load</div>;
   }
+  if (!initForecastData) return <div>loading...</div>;
+
+  const generateGeoJsonForecastData = (forecastData) => {
+    const filteredForcastData = forecastData?.forecasts?.slice(1);
+
+    const forecastGeoJson = {
+      ...gspShapeData,
+      features: gspShapeData.features.map((featureObj, index) => (
+        {
+          ...featureObj,
+          properties: {
+            ...featureObj.properties,
+            expectedPowerGenerationMegawatts: Math.floor(filteredForcastData[index].forecastValues[latestForecastValue].expectedPowerGenerationMegawatts)
+          }
+        }
+      ))
+    };
+
+    return {filteredForcastData, forecastGeoJson};
+  }
+
+  const getPaintPropsForFC = (filteredForcastData) => {
+    const allValues = filteredForcastData.map((item) => {
+      return Math.floor(item.forecastValues[latestForecastValue].expectedPowerGenerationMegawatts);
+    })
+   
+    const extent = d3.extent([...allValues]);
+
+    return {     
+      "fill-color": "#eab308",
+      "fill-opacity": 
+      [
+        "interpolate",
+        ["exponential", 1],
+        ['get', 'expectedPowerGenerationMegawatts'],
+        extent[0],
+        0,
+        extent[1],
```

Comment:
This `fill-opacity` array defines how the yellow gsp opacity levels are chosen, based on the expectedPowerGenerationMegawatts values (see https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/#interpolate).

From what I understand, this code uses the highest expectedPowerGenerationMegawatts as the highest value in the opacity scale (`extent[1]`). However in the story @peterdudfield proposed we use a fixed value of 400MW as the highest value. @Alaatohamy I believe we will need to change `extent[1]` to be `400` (probably saved in constant.ts).